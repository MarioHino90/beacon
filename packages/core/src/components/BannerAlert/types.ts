import React from "react";
import { ThemedStyledProps } from "styled-components";

import { UntransitionedBannerAlertProps } from "../UntransitionedBannerAlert";

export type Intent = "success" | "warning" | "danger" | "info";

export interface BannerAlertProps
  extends UntransitionedBannerAlertProps,
    Omit<StyledBannerTransitionProps, "children"> {}

export interface StyledBannerTransitionProps
  extends React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}

export interface IntentContextProviderProps {
  intent: Intent;
}

// this uses <any, any> because the extra props in the themed styled component
// do not matter, just that it is themed
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export interface IntentColorArgs extends ThemedStyledProps<any, any> {
  intent: Intent;
}
