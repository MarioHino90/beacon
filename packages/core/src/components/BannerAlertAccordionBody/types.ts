export interface BannerAlertAccordionBodyProps
  extends React.HTMLAttributes<HTMLDivElement> {}
