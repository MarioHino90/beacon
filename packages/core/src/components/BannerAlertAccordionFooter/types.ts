export interface BannerAlertAccordionFooterProps
  extends React.HTMLAttributes<HTMLDivElement> {}
