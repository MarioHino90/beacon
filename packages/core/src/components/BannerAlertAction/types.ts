import { ButtonProps } from "../Button";
import { Intent } from "../BannerAlert";

export interface BannerAlertActionProps
  extends Omit<ButtonProps, "variant" | "intent"> {
  /** * the visual intent */
  intent?: Intent;
}
