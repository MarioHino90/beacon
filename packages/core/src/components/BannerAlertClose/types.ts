import { Intent } from "../BannerAlert";

export interface BannerAlertCloseProps {
  /**
   * visual banner intents */
  intent: Intent;
  /**
   * on close callack */
  onClose: () => void;
}

export interface StyledProps {
  /**
   * visual banner intents */
  intent: Intent;
}
