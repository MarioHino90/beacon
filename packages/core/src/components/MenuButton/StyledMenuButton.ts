import styled, { css } from "styled-components";
import { getColor, getSpacing } from "@braze/beacon-styling";

import { StyledBodyText } from "../BodyText";
import { StyledHeading } from "../Heading";
import { StyledIcon } from "../Icon";

import { StyledMenuButtonProps } from "./types";

export const StyledMenuButton = styled.button<StyledMenuButtonProps>`
  display: flex;
  align-items: center;
  width: 100%;
  padding: ${getSpacing("sm")} ${getSpacing("xl")};
  text-align: left;

  &:hover:not(:disabled),
  &:focus:not(:disabled) {
    background-color: ${getColor("gray", 100)};
  }

  ${StyledIcon} {
    margin-right: ${getSpacing("sm")};
    font-size: 1rem;
  }

  ${(props) =>
    props.disabled &&
    css`
      cursor: not-allowed;

      ${StyledHeading}, ${StyledBodyText}, ${StyledIcon} {
        color: ${getColor("gray", 400)};
      }

      .bcl-braze-icon {
        opacity: 0.3;
      }
    `};
`;
