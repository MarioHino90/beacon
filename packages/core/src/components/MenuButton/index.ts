export { MenuButton } from "./MenuButton";

export { StyledMenuButton } from "./StyledMenuButton";

export type { MenuButtonProps, StyledMenuButtonProps } from "./types";
