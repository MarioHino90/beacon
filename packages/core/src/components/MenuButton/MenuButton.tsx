import React from "react";
import classNames from "classnames";

import { MenuButtonProps } from "./types";
import { StyledMenuButton } from "./StyledMenuButton";

export const MenuButton = ({
  className,
  ...props
}: MenuButtonProps): React.ReactElement => (
  <StyledMenuButton
    {...props}
    className={classNames(["bcl-menu-button", className])}
    type="button"
  />
);

MenuButton.displayName = "MenuButton";
