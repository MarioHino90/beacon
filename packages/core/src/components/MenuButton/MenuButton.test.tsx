import React from "react";
import { shallow } from "enzyme";
import faker from "@faker-js/faker";

import { MenuButtonProps } from "./types";
import { MenuButton } from "./MenuButton";

faker.seed(1603823212111);

describe("MenuButton", () => {
  let props: MenuButtonProps;

  beforeEach(() => {
    props = {
      children: faker.random.words(),
      className: faker.random.word(),
      id: faker.random.word(),
    };
  });

  it("should match the snapshot", () => {
    const wrapper = shallow(<MenuButton {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
