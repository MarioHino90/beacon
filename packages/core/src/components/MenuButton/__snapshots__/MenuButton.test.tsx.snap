// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton should match the snapshot 1`] = `
<StyledMenuButton
  className="bcl-menu-button dynamic"
  id="B2B"
  type="button"
>
  experiences Villages invoice
</StyledMenuButton>
`;
